Pagerduty destination doc (#1093)
rudderlabs/rudderstack-docs: docs/destinations/streaming-destinations/pagerduty.mdx

---
title: "PagerDuty"
description: Send your event data from RudderStack to PagerDuty.
aliases: ["PagerDuty destination"]
---

> [PagerDuty](https://www.pagerduty.com/) is a SaaS-based platform that helps prevent and resolve business-impacting incidents, thereby improving the customer experience.

Reviewer: ```suggestion
[PagerDuty](https://www.pagerduty.com/) is a SaaS platform that helps you prevent and resolve business-impacting incidents, thereby improving the customer experience.
```

> <div class="infoBlock">

Reviewer: ```suggestion
```

Reviewer: Reducing the no. of callouts as suggested by Pete.

> Find the open source transformer code for this destination in the <a href="https://github.com/rudderlabs/rudder-transformer/tree/master/src/v0/destinations/pagerduty">GitHub repository</a>.

Reviewer: ```suggestion
Find the open source code for this destination in the <a href="https://github.com/rudderlabs/rudder-transformer/tree/master/src/v0/destinations/pagerduty">GitHub repository</a>.
```

> </div>

Reviewer: ```suggestion
```

## Getting started

RudderStack supports sending events to PagerDuty via the following <a href="https://rudderstack.com/docs/rudderstack-cloud/rudderstack-connection-modes/">connection modes</a>:

| **Connection Mode** | **Web** | **Mobile** | **Server** |
| :--- | :--- | :--- | :--- |
| **Device mode** | -  | - | - |
| **Cloud mode** | **Supported**  | **Supported** | **Supported** |

Once you have confirmed that the source platform supports sending events to PagerDuty, follow these steps:

1. From your [RudderStack dashboard](https://app.rudderstack.com/), add a source. Then, from the list of destinations, select **PagerDuty**.
2. Assign a name to the destination and click **Continue**.

## Connection settings

To successfully configure PagerDuty as a destination, you will need to configure the following settings:

> - **PagerDuty Routing Key**: Enter the integration key from PagerDuty service. Refer to the <a href="#faq">FAQ</a> section for more information on how to obtain it.

Reviewer: ```suggestion
- **PagerDuty Routing Key**: Enter the integration key associated with your PagerDuty service. Refer to the <a href="#faq">FAQ</a> section for more information on how to obtain it.
```

Reviewer: LMK if this change makes sense.

> - **dedup_key Field Identifier**: Enter a `dedup_key` which acts as an identifier for the event. You can provide any value in this field. If not provided, PagerDuty automatically generates and assigns a `dedup_key` to the event. This key is used while updating, resolving, or acknowledging an event.

Reviewer: ```suggestion
- **dedup_key Field Identifier**: Enter a `dedup_key` which acts as the event identifier and is used while updating, resolving, or acknowledging an event. You can provide any value in this field. If not provided, PagerDuty automatically generates and assigns a `dedup_key` to the event.
```

## Track

> You can send a <Link to="/event-spec/standard-events/track">`track`</Link> call to PagerDuty to ingest the [Alert events](https://developer.pagerduty.com/docs/ZG9jOjExMDI5NTgx-send-an-alert-event) or [Change events](https://developer.pagerduty.com/docs/ZG9jOjExMDI5NTgy-send-a-change-event).

Reviewer: Slightly confused with the PagerDuty docs - they refer to `Alert events` in lower case (`alert events` as seen [here](https://developer.pagerduty.com/docs/ZG9jOjExMDI5NTgx-send-an-alert-event#:~:text=Every-,alert%20event,-has%20a%20dedup_key), but `Change Event` is written in title case. Anyway, I think the way you have presented it looks the best.

By default, the `track` call sends an alert event. However, you can specify the event type in the `integrations` object, as shown:

```javascript
integrations: {
  pagerduty: {
    type: changeEvent / alertEvent
  }
}
```

### Alert events

You can [send an alert event](https://developer.pagerduty.com/docs/ZG9jOjExMDI5NTgx-send-an-alert-event) to report a new problem, or update an ongoing problem, depending on the event type.

A sample `track` call for sending an alert event is shown below:

```javascript
rudderanalytics.track("apiSecret is not present", {
  action: "trigger",

>   dedupKey: "9116b734-7e6b-4497-ab51-c16744d4487e",

Reviewer: In the connection settings, we write this as `dedup_key` (snake case) while here it's in camel case. Please confirm if we need to update the web app, @mihir-4116 cc @ishachopra0401

  severity: "critical",
  component: "ui",
  source: "rudder-webapp",
  group: "destination",
  class: "connection settings",
  customDetails: {
    "ping time": "1500ms",
    "load avg": 0.75,
  },

>   imageURLs: [{

Reviewer: @mihir-4116 do we need all these URLs? for a sample snippet, just 1 should be fine?

| `properties.group`   | `payload.group`     | 
| `properties.class`     | `payload.class`      | 
| `properties.component`   | `payload.component`     | 
| `properties.client`   | `client`     | 
| `properties.clientUrl`     | `client_url`      | 
| `properties.customDetails`   | `payload.custom_details`     | 

### Change events

You can [send a change event](https://developer.pagerduty.com/docs/ZG9jOjExMDI5NTgy-send-a-change-event) to send information about the recent changes, such as code deployment, and system configuration changes from any system that can make an outbound HTTP connection.

A sample `track` call for sending a change event is shown below:

```javascript
rudderanalytics.track(
  "Github CI/CD Triggered", {
    source: "rudder-webapp",
    customDetails: {
      "ping time": "1500ms",
      "load avg": 0.75,
    },
    imageURLs: [{
        src: "https://static.s4be.cochrane.org/app/uploads/2017/04/shutterstock_531145954.jpg",
        alt: "first image",
      },
      {
        src: "https://chart.googleapis.com/chart?chs=600x400&chd=t:6,2,9,5,2,5,7,4,8,2,1&cht=lc&chds=a&chxt=y&chm=D,0033FF,0,0,5,1",
        alt: "second image",
      },
    ],

>     linkURLs: [{

Reviewer: Same observation as above, we can mention just one image and link URL for this sample snippet.

        text: "Js Object Error",
      },
      {
        href: "https://www.techtarget.com/whatis/definition/stack-overflow#:~:text=A%20stack%20overflow%20is%20a,been%20allocated%20to%20that%20stack",
        text: "Stack Overflow Error",
      },
    ],
  }, {
    integrations: {
      pagerduty: {
        type: "changeEvent",
      },
    },
  }
);
```

#### Property mappings

The following table lists the mappings between RudderStack and PagerDuty properties for change events:

| RudderStack property | PagerDuty property | 
| :----------------- | :--------------- |
| `event` <br/> <span style="color: #4D4DFF;font-size:12px;">Required</span>    | `payload.summary`      |
| `properties.severity`   | `payload.severity`     | 
| `properties.linkURLs`     | `links`      | 
| `properties.imageURLs`   | `images`     | 
| `originalTimeStamp`     | `payload.timestamp`      | 
| `properties.customDetails`   | `payload.custom_details`     |

## FAQ

### Where can I find the PagerDuty Routing Key?

1. Log in to the [PagerDuty dashboard](https://app.pagerduty.com/).
2. Go to **Services** > **Service Directory**.

> 3. Select your desired service and click **Integrations** tab.

Reviewer: ```suggestion
3. Select your desired service and click the **Integrations** tab.
```

> 4. Click **Add integration** and select any desired integration platform. Here, you can find the integration/routing key as shown:

Reviewer: ```suggestion
4. Click **Add integration** and select the integration platform. Here, you can find the integration/routing key as shown:
```
